Move `AggregateExpr`, `PhysicalExpr` and `PhysicalSortExpr` to physical-expr-core (#9926)
apache/arrow-datafusion: datafusion/physical-expr-core/README.md

<!---
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
-->

# DataFusion Core Physical Expressions

[DataFusion][df] is an extensible query execution framework, written in Rust, that uses Apache Arrow as its in-memory format.

> This crate is a submodule of DataFusion that provides the core functionality of physical expressions.
> Like `PhysicalExpr` or `PhysicalSortExpr` and related things.

Reviewer: Here is a very minor suggestion:

```suggestion
This crate is a submodule of DataFusion that provides shared APIs for implementing
physical expressions such as `PhysicalExpr` and `PhysicalSortExpr`.
```
